fortls/parse_fortran.py

```diff
@@ -15,6 +15,8 @@
 END_MOD_REGEX = re.compile(r'[ ]*END[ ]*MODULE', re.I)
 SUBMOD_REGEX = re.compile(r'[ ]*SUBMODULE[ ]*\(', re.I)
 END_SMOD_REGEX = re.compile(r'[ ]*END[ ]*SUBMODULE', re.I)
+BLOCK_REGEX = re.compile(r'[ ]*([a-z_][a-z0-9_]*[ ]*:)?[ ]*BLOCK', re.I)
```

Comment:
There seems to be an issue with this regex in that any line that starts with the string `block` (for instance, `blockID = 1` gets marked as the start of a `BLOCK` construct (which is then never closed) and this messes up the scopes in the file object as a result, which can lead to crashes when the end line of a block stays listed as `None`. I can open a proper issue with example code tomorrow if that would be helpful :)